Collections/Nikita/MySingleCollection.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Collections.Nikita
+{
+    class MySingleCollection<T> : Collection<T>
+    {
+        private readonly MyLogger _logger;
```

Comment:
sticking to OOP concepts, MySingleCollection should see and use only ILogger interface